waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/ThriftMetastoreClientManager.java

```diff
@@ -112,12 +110,13 @@ public void open() {
     boolean useFramedTransport = conf.getBoolVar(ConfVars.METASTORE_USE_THRIFT_FRAMED_TRANSPORT);
     boolean useCompactProtocol = conf.getBoolVar(ConfVars.METASTORE_USE_THRIFT_COMPACT_PROTOCOL);
     int clientSocketTimeout = (int) conf.getTimeVar(ConfVars.METASTORE_CLIENT_SOCKET_TIMEOUT, TimeUnit.MILLISECONDS);
+    int connectionTimeout = (int) TimeUnit.SECONDS.toMillis(2L);
```

Comment:
Maybe we can define this in this class `CloseableThriftHiveMetastoreIfaceClientFactory` and pass in an argument in this class. In `CloseableThriftHiveMetastoreIfaceClientFactory` we can add `latency` from the metastore settings, that will make it configurable.

Reply:
By default, the latency is 0, so all metastores that don't have a latency configured, would have connectionTimeout 0. So maybe connection timeout should be a different parameter.

Reply:
no same trick as you did before, we define the base + latency

Reply:
Aaah that makes sense!